### PR TITLE
slug@v5.1.0 - Adds trim property for the slug package

### PR DESCRIPTION
**Pull request by edwardofclt**
The slug package support an option to allow having whitespace at the end of a string.

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [https://github.com/Trott/slug#options](https://github.com/Trott/slug#options)
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by ddouglas — APPROVED** [review]

File: types/slug/slug-tests.ts [hunk 1]
```diff
@@ -19,6 +19,9 @@ slug("string", {}); // $ExpectType string
 
 slug("string", "replacement"); // $ExpectType string
 
+// eslint-disable-next-line no-trailing-spaces
```
Comment (martin-badin) [inline types/slug/slug-tests.ts]: remove the eslint rule

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/slug/slug-tests.ts (hunk 1 above)
Comment (edwardofclt) [reply types/slug/slug-tests.ts]: The test fails with or without the rule.

Comment (martin-badin) [reply types/slug/slug-tests.ts]: I tested it without the rule. It works.

Comment (ddouglas) [reply types/slug/slug-tests.ts]: He is trying to add functionality that is valid but the test is failing, hence the rule that he was trying to use to disable the whitespace check.

Comment (martin-badin) [reply types/slug/slug-tests.ts]: Snímka obrazovky 2021-11-29 o 21 50 45

Comment (edwardofclt) [reply types/slug/slug-tests.ts]: I mean I see that it's passing for you... but the CI tests are failing both ways.

**Comment by sandersn** [comment]
The original owner, @mhegazy, doesn't work on Typescript anymore, so it would be good to follow up with a PR that adds some new owners.
